packages/doenetml-worker-rust/lib-doenetml-core/src/core/state/types/list_marker.rs

```diff
@@ -0,0 +1,214 @@
+/// A list marker. This is used by both `<ol>` and `<ul>` elements.
+/// They are named similarly to their [HTML equivalents](https://developer.mozilla.org/en-US/docs/Web/CSS/list-style-type).
+#[derive(Debug, Clone, Copy, PartialEq, serde::Serialize, serde::Deserialize)]
+#[cfg_attr(feature = "web", derive(tsify_next::Tsify))]
+pub enum ListMarker {
+    // Ordered list
+    /// Label with a number, starting at `start`. e.g. `1`, `2`, `3`, ...
+    Decimal { start: u8 },
+    /// Label with lower case letters. e.g. `a`, `b`, `c`, ...
+    LowerAlpha,
+    /// Label with lower case roman numerals. e.g. `i`, `ii`, `iii`, ...
+    LowerRoman,
+    /// Label with upper case letters. e.g. `A`, `B`, `C`, ...
+    UpperAlpha,
+    /// Label with upper case roman numerals. e.g. `I`, `II`, `III`, ...
+    UpperRoman,
+    // Unordered lists
+    /// A filled circle.
+    Disc,
+    /// A hollow circle.
+    Circle,
+    /// A filled square.
+    Square,
+}
+
+impl ListMarker {
+    /// Get the next list marker in the sequence.
+    pub fn next_marker(&self) -> Self {
+        match self {
+            ListMarker::Decimal { .. } => Self::LowerAlpha,
+            ListMarker::LowerAlpha => Self::LowerRoman,
+            ListMarker::LowerRoman => Self::UpperAlpha,
+            ListMarker::UpperAlpha => Self::UpperRoman,
+            ListMarker::UpperRoman => Self::Decimal { start: 1 },
+            ListMarker::Disc => Self::Circle,
+            ListMarker::Circle => Self::Square,
+            ListMarker::Square => Self::Disc,
+        }
+    }
+
+    /// Convert `index` into the string representation given by this marker.
+    /// `0` maps to the number `"0"`. Lists are assumed to start indexing at `1`,
+    /// though when using `ListMarker::Decimal`, any integer value of `index` can be used.
```

Comment:
Apparently, you can use any integer value for the roman markers, as well (and will get empty strings for non-positive index for the alpha markers).

Reply:
Yes. I wasn't quite sure what to do here. Should I document this behavior? Should I allow negative letters as well?

Reply:
I'd just document it and only bother extending to negative letters if there is some use case.